numpyro/infer/util.py

```diff
@@ -427,6 +427,21 @@ def initialize_model(rng_key, model,
 
     if not_jax_tracer(is_valid):
         if device_get(~jnp.all(is_valid)):
+            with numpyro.validation_enabled(), trace() as tr:
```

Comment:
A more informative warning / error message is definitely needed. I am thinking that can we simply run `initialize_model` with `validation_enabled` (I wouldn't expect that to add any material overhead)? Is the resulting warning message not informative enough?

Reply:
Yes, it is simpler (I also don't worry about the overhead) but there are two issues with that:
+ validation is only useful for the first try (under jax loop, we can't prompt the warning/error for the later tries)
+ displaying the warning message for the first try might not be useful for users when we can find a valid one in a later try

What do you think?

Reply:
Thanks for explaining, @fehiepsi, both of your points make a lot of sense.